lms/djangoapps/courseware/course_apps.py

```diff
@@ -0,0 +1,112 @@
+"""Course app config for courseware apps."""
+from typing import Dict, Optional
+
+from django.conf import settings
+from django.contrib.auth import get_user_model
+from django.utils.translation import ugettext_noop
+from opaque_keys.edx.keys import CourseKey
+from xmodule.modulestore.django import modulestore
+
+from openedx.core.djangoapps.content.course_overviews.models import CourseOverview
+from openedx.core.djangoapps.course_apps.plugin import CourseApp
+from openedx.core.lib.courses import get_course_by_id
+
+User = get_user_model()
+
+
+class ProgressCourseApp(CourseApp):
+    """
+    Course app config for progress app.
+    """
+
+    app_id = "progress"
+    name = ugettext_noop("Progress")
+    description = ugettext_noop("Allow students to track their progress throughout the course.")
+    default_enabled = True
+
+    @classmethod
+    def is_available(cls, course_key: CourseKey) -> bool:
+        """
+        The progress course app is always available.
+        """
+        return True
+
+    @classmethod
+    def is_enabled(cls, course_key: CourseKey) -> bool:
+        """
+        The progress course enabled/disabled status is stored in the course module.
+        """
+        return not CourseOverview.get_from_id(course_key).hide_progress_tab
+
+    @classmethod
+    def set_enabled(
+        cls, course_key: CourseKey, user: User, enabled: bool
+    ) -> bool:  # pylint: disable=unused-argument
+        """
+        The progress course enabled/disabled status is stored in the course module.
+        """
+        course = get_course_by_id(course_key)
+        course.hide_progress_tab = not enabled
+        modulestore().update_item(course, user.id)
+        return enabled
+
+    @classmethod
+    def get_allowed_operations(
+        cls, course_key: CourseKey, user: Optional[User] = None
+    ) -> Dict[str, bool]:  # pylint: disable=unused-argument
+        """
+        Returns the allowed operations for the app.
+        """
+        return {
+            "enable": True,
+            "configure": True,
+        }
+
+
+class TextbooksCourseApp(CourseApp):
+    """
+    Course app config for textbooks app.
+    """
+
+    app_id = "textbooks"
+    name = ugettext_noop("Textbooks")
+    description = ugettext_noop("Provide links to applicable resources for your course.")
+    default_enabled = settings.FEATURES.get("ENABLE_TEXTBOOK")
```

Comment:
You'd use `SettingDictToggle` here. 
```
MY_FEATURE = SettingDictToggle("FEATURES", "ENABLE_TEXTBOOK" default=False, module_name=__name__)
OR
MY_FEATURE = SettingDictToggle("FEATURES", "ENABLE_TEXTBOOK" default=False)
```

Reply:
The toggles API will then consider this the source of this toggle and display that in the API. That is not what I want. In fact it is for that reason I am removing the original use of `SettingToggle` as well. I think the correct approach would be to add these to a toggles file and then import from there. However, for these course apps I would like to move these from global flags to course flags if possible, so till that approach is finalised I'll just use the raw flag directly.

Reply:
Not sure I understand all. Here is what I mean. 

```
>>> SettingDictToggle("FEATURES", "ENABLE_TEXTBOOK", default=False).is_enabled()
True
```